src/main/scala/org/tensorframes/impl/datatypes.scala

```diff
@@ -488,4 +558,69 @@ private[impl] object LongOperations extends ScalarTypeOperation[Long] with Loggi
     logTrace(s"Extracted from buffer: ${res.toSeq}")
     res
   }
-}
\ No newline at end of file
+
+  override def tag: Option[TypeTag[_]] = Option(implicitly[TypeTag[Long]])
+
+  override def ev = ClassTag.Long
+}
+
+// ********** STRING *********
+// This is actually byte arrays, which corresponds to the 'binary' type in Spark.
+
+// The string converter can only deal with one row at a time (the most common case).
+private[impl] class StringTensorConverter(s: Shape, numCells: Int)
+  extends TensorConverter[Array[Byte]](s, numCells) with Logging {
+  private var buffer: Array[Byte] = null
+
+  override val elementSize: Int = 1
+
+  {
+    logger.debug(s"Creating string buffer for shape $s and $numCells cells")
+    assert(s == Shape() && numCells == 1, s"The string buffer does not accept more than one" +
+      s" scalar of type binary. shape=$s numCells=$numCells")
+  }
+
+
+  override def reserve(): Unit = {}
+
+  override def appendRaw(d: Array[Byte]): Unit = {
+    assert(buffer == null, s"The buffer has only been set with ${buffer.length} values," +
+      s" but ${d.length} are trying to get inserted")
+    buffer = d.clone()
+  }
+
+  override def tensor2(): tf.Tensor = {
+    tf.Tensor.create(buffer)
+  }
+
+  override def fillBuffer(buff: ByteBuffer): Unit = {
+    buff.put(buffer)
+  }
+}
+
+private[impl] object StringOperations extends ScalarTypeOperation[Array[Byte]] with Logging {
+  override val sqlType = BinaryType
+  override val tfType = ProtoDataType.DT_STRING
+  override val tfType2 = tf.DataType.STRING
+  override val scalarType = ScalarBinaryType
+  final override val zero = Array.empty[Byte]
+
+  override def tfConverter(cellShape: Shape, numCells: Int): TensorConverter[Array[Byte]] =
+    new StringTensorConverter(cellShape, numCells)
+
+  override def convertTensor(t: tf.Tensor): MWrappedArray[Array[Byte]] = {
+    // TODO(tjh) implement later
+    ???
```

Comment:
Oh yes, this is a special scala syntax for holes in the program. It should be replaced by proper exceptions.